text/0140-record-feature-flags-on-error.md

```diff
@@ -0,0 +1,45 @@
+- Start Date: 2024-09-04
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/140
+- RFC Status: draft
+
+# Summary
+
+As feature flags are evaluated in a customer's application we will collect those flags and their results, store them in-memory, and push the values to Sentry when an error occurs. The flags can be used to further understand what went wrong during the session.
+
+# Motivation
+
+To enable users to debug application errors with a complete picture of their application's state.
+
+# Options Considered
+
+We will collect flag evaluations and hold them in-memory.  On error event the flags will be placed into the `contexts` object on the `event` and sent to Sentry.
+
+## Transport
+
+The flags will be represented by this data structure during transport. The `flag` key is the name of the flag and the `result` key is the evaluation result returned by the customer's application.
+
+```json
+{
+    "contexts": {
+        "flags": [
```

Comment:
Just realized that we need an object here, as Sentry won't accept an array.

Could be:

```
"flags": {
 "values": [ ]
```

Reply:
Thank you!